Mark "for each...in" as deprecated

PeteH32 opened the pull request:
<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
<!-- ✍️ In a sentence or two, describe your changes -->
The JavaScript Reference page still lists a now deprecated statement ("`for each...in`").  Rather than removing it from the list, it may be better to just add a note that it's deprecated.

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference#iterations

This page shows that the method is deprecated:

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/For-each-in_loops_are_deprecated

#### Motivation
<!-- ❓ Why are you making this change? Help us understand how your changes help readers. -->

#### Supporting details
<!-- 🔗 Link to supporting information, such as bug trackers, source control, release notes, and vendor docs. -->
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/For-each-in_loops_are_deprecated

#### Related issues
<!-- 🔨 If applicable, use "Fixes #XYZ" -->

#### Metadata
<!-- ✅ Check a box if applicable, like this: [x]

This PR…
-->
- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

<!-- 👷‍♀️ After submitting, review the results of the "Checks" tab! -->

PeteH32 committed:
Mark "for_each...in" as deprecated

wbamberg reviewed (commented):
Thanks for your PR!

But I think it would be better to remove this entry. We should also de-link all the broken links to this feature in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/For-each-in_loops_are_deprecated .

It looks like this feature is removed now (in recent Firefox and Chrome, anyway). I wonder when that happened, and whether we should also remove this error page, since it won't appear any more in browsers that have removed it. @Elchi3 , do you know anything more about this?

Elchi3 commented:
Yes, we should remove `for each in` now. The error page can go and any references to it as well. It looks like the reference page has already been deleted.

wbamberg commented:
Thanks Florian! Sounds good to me.

@PeteH32 , please let us know if you want to make this update :).

PeteH32 commented:
@wbamberg , yes, I will go ahead and update this PR with those changes.

pheirendt-jama committed:
Remove links, change "will be" to "has been"

pheirendt-jama committed:
Remove from list on reference page

PeteH32 commented:
@wbamberg , @Elchi3 ,
I made the changes you both mentioned.

- Removed `for each in` from the JS Reference page
- Removed link to page from `for in` page
- Updated the page about error "for-each-in loops are deprecated" so that instead of  saying "will be removed" it now says "has been removed"
  - I did not remove this error page entirely, since it is the only thing left documenting that this once did exist. I'm not sure if that is the better option, so just let me know if you want the entire page removed

wbamberg commented:
Thank you @PeteH32 !

I think we should also delete https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/For-each-in_loops_are_deprecated . We don't generally keep historical information on MDN.

You can delete pages using `yarn content delete` (https://github.com/mdn/content#deleting-a-document). You should also delete the link to it in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors.

I can't think of a good page to use as a redirect, so no need to create one.

Thanks again for your help with this!

wbamberg reviewed (approved):
Thank you @PeteH32 for this fix. I'll remove the error page in a follow-up.
